notebooks/Binary Classification with the UCI Credit Card Payment Default Dataset.ipynb

```diff
@@ -0,0 +1,599 @@
+    "# Scores on test set\n",
+    "test_scores = model.predict_proba(df_test)[:, 1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Train AUC\n",
+    "roc_auc_score(Y_train, model.predict_proba(df_train)[:, 1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Predictions (0 or 1) on test set\n",
+    "test_preds = (test_scores >= np.mean(Y_train)) * 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# LightGBM feature importance \n",
+    "lgb.plot_importance(model, height=0.6, title=\"Features importance (LightGBM)\", importance_type=\"gain\", max_num_features=15) \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We notice that the biased feature `LIMIT_BAL`, appears as the most important feature in this model although it has no predictive power for an entire demographic segment in the data. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Helper functions\n",
+    "def group_equalized_odds_diff(y_true, y_pred, group):\n",
+    "    TPR_diff = group_recall_score(y_true, y_pred, group).range\n",
+    "    TNR_diff = group_recall_score(1-y_true, 1-y_pred, group).range\n",
+    "    return max(TPR_diff, TNR_diff)\n",
+    "\n",
+    "def get_metrics_df(models_dict, y_true, group):\n",
+    "    metrics_dict = {\n",
+    "        \"Demographic parity difference\": (lambda x: group_selection_rate(y_true, x, group).range, True),\n",
+    "        \"Demographic parity ratio\": (lambda x: group_selection_rate(y_true, x, group).range_ratio, True),\n",
+    "        \"Error rate difference\": (\n",
+    "            lambda x: metric_by_group(balanced_accuracy_score, y_true, x, group).range, True),\n",
+    "        \"Equal opportunity difference\": (lambda x: group_recall_score(y_true, x, group).range, True),\n",
+    "        \"Equalized odds difference\": (lambda x: group_equalized_odds_diff(y_true, x, group), True),\n",
+    "        \"Group AUC difference\": (lambda x: group_roc_auc_score(y_true, x, group).range, False),\n",
+    "        \"Overall AUC\": (lambda x: roc_auc_score(y_true, x), False)\n",
+    "        }\n",
+    "    df_dict = {}\n",
+    "    for metric_name, (metric_func, use_preds) in metrics_dict.items():\n",
+    "        df_dict[metric_name] = [metric_func(preds) if use_preds else metric_func(scores) \n",
+    "                                for model_name, (preds, scores) in models_dict.items()]\n",
+    "    return pd.DataFrame.from_dict(df_dict, orient=\"index\", columns=models_dict.keys())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We calculate several disparity metrics below:"
```

Comment:
Should we highlight some things about the table? For example, you show the error rate difference, but I don't see the overall error rate for comparison (while a change of 17% is always going to be large, 1% vs 18% is much more significant than 80% vs 90%.

Also, I don't believe that you mentioned if the underlying dataset has a gender imbalance?

Reply:
addressed